Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -1336,6 +1392,110 @@ bool IsTracking() const {
             // ARTrackingStateNormal.
             return SystemImpl.XrContext->Frame.camera.trackingState == ARTrackingState::ARTrackingStateNormal;
         }
+        
+        std::vector<ImageTrackingScore>* GetImageTrackingScores() {
+            if (imageTrackingScoresValid) {
+                return &imageTrackingScores;
+            } else {
+                return nil;
+            }
+        }
+        
+        void CreateAugmentedImageDatabase(const std::vector<ImageTrackingRequest>& requests) {
+            // Create a dispatch group for the validation tasks.
+            __block dispatch_queue_t validationQueue{dispatch_queue_create("Image Validation Queue", nil)};
```

Comment:
We already use arcana's task system in this file for other types of asynchrony. Could we use arcana here as well instead of using all the dispatch stuff? It seems like it would be less of a maintenance burden because people only need to understand one system (arcana) rather than two systems, plus it would be pure c++ and not use a bunch of obj-c. Thoughts?

Reply:
I'll give it a shot